### PR TITLE
fix(jira): Hide fields that we want people to set in jira

### PR DESCRIPTION
**Pull request by macqueen**

### CODE REVIEW

File: src/sentry/integrations/jira/integration.py [hunk 1]
```diff
@@ -394,13 +396,19 @@ def get_create_issue_config(self, group, **kwargs):
 
         # TODO(jess): are we going to allow ignored fields?
         # ignored_fields = (self.get_option('ignored_fields', group.project) or '').split(',')
-        ignored_fields = set()
+
+        # hide sprint, epic link, parent and linked issues fields because they don't work
+        # since sprint and epic link are "custom" we need to search fo rthem by name
+        ignored_fields = set(
+            k for k, v in six.iteritems(issue_type_meta['fields']) if v['name'] in {'Sprint', 'Epic Link'}
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/jira/integration.py]: ooo, rare set syntax.

Can you suck this out into a constant at the top

`HIDDEN_ISSUE_FIELDS= {'Sprint', 'Epic Link'}`

File: src/sentry/integrations/jira/integration.py [hunk 2]
```diff
@@ -394,13 +396,19 @@ def get_create_issue_config(self, group, **kwargs):
...
+        )
+        ignored_fields.update(['parent', 'issuelinks'])
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/jira/integration.py]: Can you document why we have to do this differently for the different files (sprint and epic vs parent and issue link)

Comment (macqueen) [reply src/sentry/integrations/jira/integration.py]: i did: https://github.com/getsentry/sentry/pull/9421/files#diff-f78e256a535cb9c8f27ccbaa1f1f4ac8R401

File: src/sentry/integrations/jira/integration.py (hunk 1 above)
Comment (macqueen) [reply src/sentry/integrations/jira/integration.py]: yeah, i thought about doing this, but it's also sort of weird that then i'd also have to have another global for the ones we can should use keys for :-/

File: src/sentry/integrations/jira/integration.py (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/integrations/jira/integration.py]: I can't read lol

File: src/sentry/integrations/jira/integration.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/integrations/jira/integration.py]: how about 

```py
HIDDEN_ISSUE_FILEDS = {
  'by_key': ['parent', 'issuelinks']
  'by_custom_name': ['Sprint', 'Epic Link'],
}
```

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**